simulator.py

```diff
@@ -141,6 +148,64 @@ def __init__(self, env, name, age, rng, infection_timestamp, household, workplac
     def __repr__(self):
         return f"H:{self.name}, SEIR:{int(self.is_susceptible)}{int(self.is_exposed)}{int(self.is_infectious)}{int(self.is_removed)}"
 
+    def handle_message(self, m_i):
+        m_i_enc = _encode_message(m_i)
+        temp_cur_num_messages = self.cur_num_messages
+        # if first message received, it's group 0
+        if self.cur_num_messages == 0:
+            self.A[m_i_enc] = 0
+            self.M.append(m_i_enc)
+            self.cur_num_messages += 1
+            return
+
+        scores = {}
+        for m_enc in self.M:
+            m = _decode_message(m_enc)
+            if m_i[0] == m[0] and m_i[2].day == m[2].day:
+                scores[m_enc] = 3
+            elif m_i[0][:3] == m[0][:3] and m_i[2].day-1 == m[2].day:
+                scores[m_enc] = 2
+            elif m_i[0][:2] == m[0][:2] and m_i[2].day - 2 == m[2].day:
+                scores[m_enc] = 1
+            elif m_i[0][:1] == m[0][:1] and m_i[2].day - 2 == m[2].day:
+                scores[m_enc] = 0
+
```

Comment:
Just thinking if this operation can be just brought down to row-wise multiplication of the new message (a 0-1 array) with the previous message (a number of past messages x number of bits) array. The best match will be the row with the maximum sum. I am guessing there will be a lot of ties there. Should we randomize it (loose a lot of information) or use all of them?

Reply:
this is an interesting idea, and we should pursue improvements on the disambiguation / clustering code. However, I think for this PR it is a sufficient draft that yields 57% accuracy. A weak but acceptable baseline.